src/main.rs

```diff
@@ -1,28 +1,63 @@
-fn main() {
-    println!("😃_>> Olá, gostaria de adicionar um novo TODO? (s/n) ");
-    loop {
-        let answer = input();
-        if answer == "s" {
-            add_todo();
-            println!("😃_>> Gostaria de adicionar outro TODO? (s/n) ");
-        } else if answer == "n" {
-            println!("😃_>> Obrigado por usar o TODO-CLI! 👋");
-            break;
-        } else {
-            println!("🤨_>> Desculpa eu não entendi. Digite 's' se deseja adicionar um novo TODO ou 'n' se deseja sair. ");
-        }
+use std::io::{Stdin, Stdout, Write};
+
+#[derive(Debug, Clone)]
+struct Todo {
+    message: String,
+}
+
+impl Todo {
+    fn new(message: String) -> Self {
+        Self { message }
     }
 }
 
-fn add_todo() {
-    println!("😃 >> Qual é o TODO?");
-    let todo = input();
-    println!("\n😃_>> O TODO foi adicionado com sucesso! \n");
-    println!("📝 - {} \n", todo);
+struct Terminal {
+    stdin: Stdin,
+    stdout: Stdout,
+}
+
+impl Terminal {
+    fn new() -> Self {
+        Self {
+            stdin: std::io::stdin(),
+            stdout: std::io::stdout(),
+        }
+    }
+
+    fn input(&mut self) -> String {
+        let mut buf = String::new();
+        self.stdin.read_line(&mut buf).unwrap();
+        buf.trim().to_string()
+    }
+
+    fn ask_for_new_todo(&mut self) -> Todo {
+        println!("😃_>> Olá, gostaria de adicionar um novo TODO? (s/n) ");
+        loop {
+            let answer = self.input();
+            if answer == "s" {
+                println!("😃 >> Qual é o TODO?");
+                let message = self.input();
+                let todo = Todo::new(message);
+                return todo;
```

Comment:
Normalmente não se cria uma variável apenas para retorna-la.

```suggestion
                return Todo::new(message);
```

Reply:
Verdade... fica bem mais limpo assim. Valeu mesmo